docs/sphinx/examples.rst

```diff
@@ -22,6 +22,7 @@ powerful |hpx| concepts.
 .. toctree::
    :maxdepth: 1
 
+   examples/matrix_multiplication
    examples/fibonacci_local
```

Comment:
```suggestion
   examples/fibonacci_local
   examples/matrix_multiplication
```

Putting plain async first would be my preference, but I can see the case for the other way around as well. @hkaiser, @aurianer, @gonidelis, @NK-Nikunj preferences?